analyzers/tests/SonarAnalyzer.Test/Rules/MessageTemplatesShouldBeCorrectTest.cs

```diff
@@ -0,0 +1,139 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.Rules.CSharp;
+
+namespace SonarAnalyzer.Test.Rules;
+
+[TestClass]
+public class MessageTemplatesShouldBeCorrectTest
+{
+    private static readonly IEnumerable<MetadataReference> LoggingReferences =
+        NuGetMetadataReference.MicrosoftExtensionsLoggingAbstractions()
+        .Concat(NuGetMetadataReference.NLog(Constants.NuGetLatestVersion))
+        .Concat(NuGetMetadataReference.Serilog(Constants.NuGetLatestVersion));
+
+    private static readonly VerifierBuilder Builder = new VerifierBuilder<MessageTemplatesShouldBeCorrect>()
+        .AddReferences(LoggingReferences);
+
+    [TestMethod]
+    public void MessageTemplatesShouldBeCorrect_CS() =>
+        Builder.AddPaths("MessageTemplatesShouldBeCorrect.cs").Verify();
+
+    [DataTestMethod]
+    [DataRow("LogCritical")]
+    [DataRow("LogDebug")]
+    [DataRow("LogError")]
+    [DataRow("LogInformation")]
+    [DataRow("LogTrace")]
+    [DataRow("LogWarning")]
+    public void MessageTemplatesShouldBeCorrect_MicrosoftExtensionsLogging_CS(string methodName) =>
+        Builder.AddSnippet($$$"""
+            using System;
+            using Microsoft.Extensions.Logging;
+
+            public class Program
+            {
+                public void Method(ILogger logger, string user, int cnt)
+                {
+                    Console.WriteLine("Login failed for {User", user);                  // Compliant
+                    logger.{{{methodName}}}("Login failed for {User}", user);           // Compliant
+
+                    logger.{{{methodName}}}("{", user);                                 // Noncompliant {{Log message template should be syntactically correct.}}
+                    logger.{{{methodName}}}("Login failed for {}", user);               // Noncompliant {{Log message template should not contain empty placeholder.}}
+                    logger.{{{methodName}}}("Login failed for {%User}", user);          // Noncompliant {{Log message template placeholder '%User' should only contain chars, numbers, and underscore.}}
+                    logger.{{{methodName}}}("Retry attempt {Cnt,r}", cnt);              // Noncompliant {{Log message template placeholder 'Cnt' should have numeric alignment instead of 'r'.}}
+                    logger.{{{methodName}}}("Retry attempt {Cnt:}", cnt);               // Noncompliant {{Log message template placeholder 'Cnt' should not have empty format.}}
```

Comment:
It would be useful to have a test case with the logging method used as a static method instead of extension.